packages/react/src/components/uiComponents.tsx

```diff
@@ -185,14 +230,14 @@ export const SignUp = withClerk(({ clerk, ...props }: WithClerkProp<SignUpProps>
   );
 }, 'SignUp');
 
-export function UserProfilePage({ children }: PropsWithChildren<UserProfilePageProps>) {
+export function UserProfilePage(_: PropsWithChildren<UserProfilePageProps>) {
   logErrorInDevMode(userProfilePageRenderedError);
-  return <>{children}</>;
+  return null;
 }
 
-export function UserProfileLink({ children }: PropsWithChildren<UserProfileLinkProps>) {
+export function UserProfileLink(_: PropsWithChildren<UserProfileLinkProps>) {
   logErrorInDevMode(userProfileLinkRenderedError);
-  return <>{children}</>;
+  return null;
```

Comment:
Types cannot be changed as it would be breaking. But `children` were never used so developers do not have the expectation if them seeing children.

I've added e2e tests cases to handle this and not leak children unless devs opt-in to the introduced feature.